github/GithubException.py

```diff
@@ -106,6 +106,57 @@ def __str__(self) -> str:
 class BadCredentialsException(GithubException):
     """
     Exception raised in case of bad credentials (when Github API replies with a 401 or 403 HTML status)
+
+    ### Possible Causes
+
+    This exception is raised possibly because of:
+    1. Insufficient scope of token
+    2. An expired token. In an application with **cache** mechanism, the cached PyGithub-like objects might
+    contain an expired token unexpectedly. ATTENTION: The PyGithub-like objects can not update the token
+    automatically.
+
+    ### One way to find the cause
+    
+    One way to find the root cause is as follows.
+    The steps are abstracted from https://github.com/PyGithub/PyGithub/issues/1753#issuecomment-1039533679
+    
+    1. In the source code directory of PyGithub package downloaded by
+    pip(often `lib/python3.x/site-packages/` in virtual env or `~/.local/lib/python3.x/site-packages/`), 
+    edit `github/Requester.py`.
+    
+    replace in `def __log`:
```

Comment:
```suggestion
    1. Identify the location of the github source code installed via pip or virtual env:
    
    ```bash
    python -c "from github import Requester; print(Requester.__file__)"
    ```

    2. Edit this `github/Requester.py` file and replace in `def __log`:
```